src/tests/end-to-end/utils/utils.h

```diff
@@ -4,24 +4,189 @@
 
-Antares::Data::Area* addArea(Antares::Data::Study::Ptr pStudy, const std::string& areaName, int nbTS);
 
-std::shared_ptr<Antares::Data::ThermalCluster> addCluster(Antares::Data::Area* pArea, const std::string& clusterName, double maximumPower, double cost, int nbTS, int unitCount = 1);
+void initializeStudy(Study::Ptr study);
+void configureLinkCapacities(AreaLink* link);
 
-std::shared_ptr<Antares::Data::BindingConstraint> addBindingConstraints(Study &study, const std::string& name, const std::string& group);
 
-void cleanSimulation(Antares::Solver::Simulation::ISimulation< Antares::Solver::Simulation::Economy >* simulation);
-void cleanStudy(Antares::Data::Study::Ptr pStudy);
+template<class MatrixType>
+class TimeSeriesConfigurer
+{
+public:
+    TimeSeriesConfigurer() = default;
+    TimeSeriesConfigurer(MatrixType& matrix) : ts_(&matrix) {}
+    TimeSeriesConfigurer& setColumnCount(unsigned int columnCount);
+    TimeSeriesConfigurer& fillColumnWith(unsigned int column, double value);
+private:
+    MatrixType* ts_ = nullptr;
+};
+
+template<class MatrixType>
+TimeSeriesConfigurer<MatrixType>& TimeSeriesConfigurer<MatrixType>::setColumnCount(unsigned int columnCount)
+{
+    ts_->resize(columnCount, HOURS_PER_YEAR);
+    return *this;
+}
+
+template<class MatrixType>
+TimeSeriesConfigurer<MatrixType>& TimeSeriesConfigurer<MatrixType>::fillColumnWith(unsigned int column, double value)
+{
+    ts_->fillColumn(column, value);
+    return *this;
+}
+
+
+class ThermalClusterConfig
+{
+public:
+    ThermalClusterConfig() = default;
+    ThermalClusterConfig(ThermalCluster* cluster);
+    ThermalClusterConfig& setNominalCapacity(double nominalCapacity);
+    ThermalClusterConfig& setUnitCount(unsigned int unitCount);
+    ThermalClusterConfig& setCosts(double cost);
+    ThermalClusterConfig& setAvailablePowerNumberOfTS(unsigned int columnCount);
+    ThermalClusterConfig& setAvailablePower(unsigned int column, double value);
+
+private:
+    ThermalCluster* cluster_ = nullptr;
+    TimeSeriesConfigurer<Matrix<double>> tsAvailablePowerConfig_;
+};
+
+std::shared_ptr<ThermalCluster> addClusterToArea(Area* area, const std::string& clusterName);
+void addScratchpadToEachArea(Study& study);
+
+// -------------------------------
+// Simulation results retrieval
+// -------------------------------
+class averageResults
+{
+public:
+    averageResults(Variable::R::AllYears::AverageData& averageResults) : averageResults_(averageResults)
+    {}
+
+    double hour(unsigned int hour) { return averageResults_.hourly[hour]; }
+    double day(unsigned int day) { return averageResults_.daily[day]; }
+    double week(unsigned int week) { return averageResults_.weekly[week]; }
+
+private:
+    Variable::R::AllYears::AverageData& averageResults_;
+};
+
+
+class OutputRetriever
+{
+public:
+    OutputRetriever(std::shared_ptr<ISimulation<Economy>>& simulation) : simulation_(simulation) {}
+    averageResults overallCost(Area* area);
```

Comment:
Not very important as long as we are in tests, but it would be better if we ask arguments as reference since it should not be null, and it should probably be const because I don't expect it to modifiy the objects (Area and so on).